internal/controller/api_to_config_test.go

```diff
@@ -902,11 +902,297 @@ func TestConversion(t *testing.T) {
 			},
 			err: nil,
 		},
+		{
+			name: "Multiple Routers and Neighbors, Multiple FRRConfigurations",
+			fromK8s: []v1beta1.FRRConfiguration{
+				{
+					Spec: v1beta1.FRRConfigurationSpec{
+						BGP: v1beta1.BGPConfig{
+							Routers: []v1beta1.Router{
+								{
+									ASN: 65010,
+									ID:  "192.0.2.5",
+									Neighbors: []v1beta1.Neighbor{
+										{
+											ASN:     65012,
+											Address: "192.0.2.7",
+											Port:    179,
+											ToAdvertise: v1beta1.Advertise{
+												Allowed: v1beta1.AllowedPrefixes{
+													Prefixes: []string{"192.0.2.10/32", "192.0.2.11/32"},
+													Mode:     v1beta1.AllowRestricted,
+												},
+												PrefixesWithCommunity: []v1beta1.CommunityPrefixes{
+													{
+														Community: "10:100",
+														Prefixes:  []string{"192.0.2.10/32"},
+													},
+													{
+														Community: "10:101",
+														Prefixes:  []string{"192.0.2.10/32", "192.0.2.11/32"},
+													},
+												},
+												PrefixesWithLocalPref: []v1beta1.LocalPrefPrefixes{
+													{
+														LocalPref: 200,
+														Prefixes:  []string{"192.0.2.10/32"},
+													},
+												},
+											},
+										},
+									},
+									VRF:      "",
+									Prefixes: []string{"192.0.2.10/32", "192.0.2.11/32"},
+								},
+								{
+									ASN: 65013,
+									Neighbors: []v1beta1.Neighbor{
+										{
+											ASN:     65017,
+											Address: "192.0.2.7",
+											Port:    179,
+											ToAdvertise: v1beta1.Advertise{
+												Allowed: v1beta1.AllowedPrefixes{
+													Prefixes: []string{"192.0.2.5/32"},
+													Mode:     v1beta1.AllowRestricted,
+												},
+											},
+										},
+									},
+									VRF:      "vrf2",
+									Prefixes: []string{"192.0.2.5/32"},
+								},
+								{
+									ASN: 65013,
+									ID:  "2001:db8::3",
+									Neighbors: []v1beta1.Neighbor{
+										{
+											ASN:     65014,
+											Address: "2001:db8::4",
+											Port:    179,
+											ToAdvertise: v1beta1.Advertise{
+												Allowed: v1beta1.AllowedPrefixes{
+													Mode: v1beta1.AllowAll,
+												},
+											},
+										},
+									},
+									VRF:      "vrf2",
```

Comment:
is this a realistic example?
Same vrf for two entries of the same config. I would even block this when we add webhooks

Reply:
nope probably mis-copy pasted, and I agree that we should be blocking these kind of stuff